rails/app/controllers/search_controller.rb

```diff
@@ -356,32 +357,29 @@ def add_material_to_collections
       assign_summary_data << [collection.name, newly_assigned_material_names,already_assigned_material_names]
     end
 
-    if request.xhr?
-      render :update do |page|
-        materials = []
-        if runnable_ids.length == 1
-          if runnable_type == "Investigation"
-            materials.push ::Investigation.find(params[:material_id])
-          elsif runnable_type == "Activity"
-            materials.push ::Activity.find(params[:material_id])
-          elsif runnable_type == "ExternalActivity"
-            materials.push ::ExternalActivity.find(params[:material_id])
-          end
-        else
-          runnable_ids.each do |id|
-            materials.push ::Activity.find(id)
-          end
-        end
-
-        if collection_ids.count > 0
-          material_names = materials.map {|m| "<b>#{m.name}</b>" }.join(", ").gsub("'","\\'")
-          page << "close_popup()"
-          page << "getMessagePopup('<div class=\"feedback_message\">#{material_names} #{'is'.pluralize(runnable_ids.length)} assigned to the selected collection(s) successfully.</div>', #{skip_reload})"
-        else
-          page << "$('error_message').update('Select at least one collection to assign this #{runnable_type}');$('error_message').show()"
-        end
+    materials = []
+    if runnable_ids.length == 1
+      if runnable_type == "Investigation"
+        materials.push ::Investigation.find(params[:material_id])
+      elsif runnable_type == "Activity"
+        materials.push ::Activity.find(params[:material_id])
+      elsif runnable_type == "ExternalActivity"
+        materials.push ::ExternalActivity.find(params[:material_id])
+      end
+    else
+      runnable_ids.each do |id|
+        materials.push ::Activity.find(id)
```

Comment:
Since we can't add Activities to collections anymore. I'd remove this, and add a check that runnable_ids has a length of 1 otherwise raise an exception.